lib/s3routes/routesUtils.ts

```diff
@@ -408,15 +411,18 @@ function retrieveData(
         client, implName, config, kms, metadata, locStorageCheckFn, vault);
     return eachSeries(locations,
         (current, next) => data.get(current, response, log,
-            (err: any, readable: http.IncomingMessage) => {
+            (err: Error, readable: http.IncomingMessage) => {
                 const cbOnce = jsutil.once(next);
                 // NB: readable is of IncomingMessage type
                 if (err) {
-                    log.error('failed to get object', {
+                    // In this case, we do not destroy the response:
+                    // nothing was sent to the client yet, so we
+                    // can return a proper error.
+                    log.debug('failed to get object', {
```

Comment:
nit: not sure about the log level change. This "inner" log has all the context of the error, so may be more accurate... But at the same time indeed the final log is sure to be called, in every situation.

not sure which way to go, but if we change this error then the other logs must be consistent: i.e. `log.error('error piping data from source'` should also be changed to lower log level.

Reply:
I kept the error log, better dump twice than missing an error... And it remains consistent with other logs.